### PR TITLE
Private Events Application

### PR DESCRIPTION
**Pull request by tacodtripe**
In this application, I was required to exercise the basic relations between models on RoR.
The features offered are:
- Users can create their own events
- Users can attend to others user's events

### CODE REVIEW

File: app/views/events/index.html.erb
Comment (oloomoses) [inline app/views/events/index.html.erb]: - [ ] Ensure to move this logic into the helpers

File: app/views/events/show.html.erb
Comment (oloomoses) [inline app/views/events/show.html.erb:36]: - [ ] Ensure to move this logic into the helpers

File: app/views/layouts/application.html.erb
Comment (oloomoses) [inline app/views/layouts/application.html.erb:24]: - [ ] Ensure to move this logic into the helpers

**Review by oloomoses — CHANGES_REQUESTED** [review]
# Status: Changes Required
Hello @tacodtripe, A good start on this project :muscle: 
Some changes to be made before its approval :point_down: 
- [ ] Kindly Ensure that you add a professional readme for this project
- [ ] Ensure that you move all your logic from the views into the helpers
- [ ] Event's index page displays past and upcoming events lists  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- [ ] The user's show page displays events the user has created and events that the user is attending to separated in past and upcoming([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- [ ] Ensure that Navigation links are present at the top of the page ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/navigation_links_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
#### Models
##### Event
- [ ] Kindly implement the scopes for upcoming and previous events [Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/scopes_requirement.png)
#### Associations
##### User as event creator
* [ ] User model: `has_many :events, foreign_key: :creator_id` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_assoc_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [ ] Event model `belongs_to :creator, class_name: "User"` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_assoc_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
#### Controllers
##### EventsController
###### `index` action
* [ ] lists past events, i.e. `@past_events = Event.previous_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [ ] lists ucomping events, i.e. `@upcoming_events = Event.upcoming_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

##### UsersController
###### `show` action
* [ ] lists events created by user, i.e. `@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [ ] lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [ ] lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

#### Tests
- [ ] Add Unit and Intergration tests

Kindly make all the above changes before submitting another review.

### Feel free to reach me on slack [Oloo Moses] incase of doubts or concerns

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Changes Required 🙏🏿 
Nice work fixing some of the features as pointed out by the previous reviewer. A user can now create events, see future and past events, login, logout. Well done!

Let's work on the following so we can merge this PR.
- [ ] You still don't have a professional README file. README's give contributors an overview of what the project is about and also makes your repo look professional. Use this [template](https://github.com/microverseinc/readme-template)
- [ ] Add unit tests for the app's logic especially for model validations,  events creation, past and future events, etc. Also instructions for running the tests in your README file.

### Martin Afani

**Review by mke2111 — CHANGES_REQUESTED** [review]
# **STATUS: CHANGES REQUIRED 🔴 ⚠️** 

## Hey @whiz25, 👋🏽 
 **Thank you for the good job you got going on and implementing the requested changes. The App works just fine.**

## REQUIRED CHANGES

- [ ] Kindly add concise and clear instructions for the cloning procedure. `bundle install` is a very crucial line of code that needs to run before you start the server.
- [ ] Kindly remove all logic from the view files. All conditional statements and loops should be removed from the views folder and transferred to the helpers folder.


- [ ] authentication should have integration tests (Rspec + Capybara) [Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/)
- [ ] event managment features should have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))


- [ ] [OPTIONAL] Adding styles to the app will give it a sleek UI, give it a second thought. 🥺 

**Kindly make all the requested changes and submit another CRR. 👍🏽** 

## Cheers!! 🍹 & Happy Coding.

# Reach me on slack for a chitchat 🆘  **Roy Mukuye**

**Review by NtwaliHeritier — CHANGES_REQUESTED** [review]
### **STATUS: CHANGES REQUESTED🛑🛑🛑🛑**

Hello @tacodtripe,
Great work on this project and making the requested changes.
However there are final changes before approving it.
I like the testing part👌

[REQUIRED]: As requested by the TSE, please make your readme file more descriptive to make it easy for people to use your app in their local machine.

That's it. Make the required changes and ask for another review👍👍
Happy coding 💻

**Review by ghost — APPROVED** [review]
# Status: Approved 🎉 

**Hello team 👋, before you merge, add in your README file instruction manual that the user should run `rails db:create` and `rails db:migrate` to set up the migrations.**

**Congratulations 👏, and good luck on your next project 😃.**

**Pull request merged**
